### ISSUE TITLE
unable to install Test::MockModule using either cpanm or cpm

### ISSUE THREAD

**Issue by XSven** [geofffranks/test-mockmodule#63]
perl 5.14.4
Test::MockModule v0.178.0

I am not able to install `Test::MockModule` using either `cpanm` or `cpm`. The unit tests can't locate _Test/Warnings.pm_. This issue isn't new it happened with the previous version v0.177.0 too. Please simply use one of the CPAN clients to reproduce it or as an alternative use this:

1. clone the git repository
2. `perl Build.PL`
3. `perl Build installdeps`
4. `perl Build test` will fail

**Comment by atoomic** [comment]
Thanks for the report, I need to have a closer look at it
I think the current CI workflow is biased as it s run on a snapshot with many CPAN distro pre-installed.

**Comment by atoomic** [comment]
Note that `Test::Warnings` is listed as a test dependency here
https://github.com/geofffranks/test-mockmodule/blob/a2d81082e1c843f9928b662645f6b4ea1df8bcdb/Build.PL#L29

I tried to reproduce the issue you described there without success

1. Removing the `Test::Warnings` package file itself (and also Test::MockModule)
```
rm -f $(perldoc -lm Test::Warnings); rm -f $(perldoc -lm Test::MockModule)
```
2. Then try to install the distro using cpanm
```
> cpanm Test::MockModule
Successfully installed Test-Warnings-0.033
Successfully installed Test-MockModule-v0.178.0
2 distributions installed
```

Note that it reinstalled the missing dependency used for testing.
Could you provide more informations about the error you ran into?

thanks

**Comment by XSven** [comment]
perl 5.16.3
cpanm 1.7047
Module::Build (0.39_01 <= THIS IS THE ISSUE)

Extract from cpanm log
```
...
Unpacking Test-MockModule-v0.178.0.tar.gz
Entering Test-MockModule-v0.178.0
Checking configure dependencies from META.json
Checking if you have Module::Build 0.38 ... Yes (0.39_01)
Checking if you have ExtUtils::Install 1.46 ... Yes (2.06)
Configuring Test-MockModule-v0.178.0
Running Build.PL
Checking prerequisites...
  requires:
    !  SUPER is not installed

ERRORS/WARNINGS FOUND IN PREREQUISITES.  You may wish to install the versions
of the modules indicated above before proceeding with this installation

Created MYMETA.yml and MYMETA.json
Creating new 'Build' script for 'Test-MockModule' version 'v0.178.0'
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Scalar::Util 0 ... Yes (1.25)
Checking if you have SUPER 1.20 ... No
Checking if you have Carp 0 ... Yes (1.26)
...
```

Now I have checked _MYMETA.json_

```
   "prereqs" : {
      "configure" : {
         "requires" : {
            "Module::Build" : "0.38"
         }
      },
      "runtime" : {
         "requires" : {
            "Carp" : "0",
            "SUPER" : "1.20",
            "Scalar::Util" : "0",
            "perl" : "5.006"
         }
      }
   },
```

The test dependencies are missing! If I am using the most recent `Module::Build` version (0.4234) I am able to install `Test::MockModule` using `cpanm`. Conclusion the minimum version requirement for `Module::Build` has to be fixed.

**0.39_01** is `Module::Build` DEV release. I do not understand why this is part of a stable perl 5.16.3 perl?! I would suggest to set the minimum version requirement for `Module::Build` to **0.4008** instead of **0.38**.

**Comment by atoomic** [comment]
Thanks for these extra details it clearly makes sense.

What would be wrong in bumping the requirement for `Module::Build` to `0.4234`?
Many fixes have been provided since `0.4008` and this would not impact the Perl version support

[ChangeLog ](https://metacpan.org/dist/Module-Build/changes)

**Comment by XSven** [comment]
Of course using the most recent version of `Module::Build` as minimum version requirement would be ok too. Please go ahead and decide.
